src/components/cc-domain-management/cc-domain-management.js

```diff
@@ -0,0 +1,819 @@
+      domainFormState: { type: Object, attribute: 'domain-form-state' },
+      domainListState: { type: Object, attribute: 'domain-list-state' },
+      _sortedDomains: { type: Array, state: true },
+    };
+  }
+
+  /**
+   * @returns {DomainManagementFormStateIdle}
+   */
+  static get INIT_DOMAIN_FORM_STATE () {
+    return {
+      type: 'idle',
+      hostname: {
+        value: '',
+      },
+      pathPrefix: {
+        value: '',
+      },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {DomainManagementDnsInfoState} Sets the state of the DNS info section */
+    this.dnsInfoState = { type: 'loading' };
+
+    /** @type {DomainManagementFormState} Sets the state of the form section */
+    this.domainFormState = CcDomainManagement.INIT_DOMAIN_FORM_STATE;
+
+    /** @type {DomainManagementListState} Sets the state of the domain list section. */
+    this.domainListState = { type: 'loading' };
+
+    /** @type {RefCcInputText} */
+    this._domainInputRef = createRef();
+
+    /** @type {RefHTMLParagraphElement} */
+    this._emptyMessageRef = createRef();
+
+    /** @type {FormattedDomainInfo[]|null} */
+    this._sortedDomains = null;
+
+    new LostFocusController(this, '.delete-domain', ({ suggestedElement }) => {
+      if (suggestedElement != null) {
+        suggestedElement.focus();
+      }
+      else {
+        this._emptyMessageRef.value?.focus();
+      }
+    });
+
+    new LostFocusController(this, '.mark-primary', () => {
+      focusBySelector(this.shadowRoot, '.primary');
+    });
+  }
+
+  /**
+   * @param {string} hostnameValue
+   * @returns {FormError|null} The error code or `null` if no error
+   */
+  _getErrorCode (hostnameValue) {
+    try {
+      const { pathname } = parseDomain(hostnameValue?.trim());
+
+      if (!hostnameValue.match(/.+\..+/)) {
+        return 'invalid-format';
+      }
+
+      if (pathname !== '/') {
+        /** We store the path so we can tell users to move this part to the "" input field instead. */
+        return { code: 'hostname-contains-path', pathWithinHostname: pathname };
+      }
+
+      return null;
+    }
+    catch (error) {
+      if (error instanceof DomainError) {
+        return error.message;
+      }
+
+      return 'invalid-format';
+    }
+
+  }
+
+  /**
+   * @param {FormError} error
+   * @returns {string|null}
+   */
+  _getErrorMessage (error) {
+    if (error === 'empty') {
+      return i18n('cc-domain-management.form.domain.error.empty');
+    }
+    if (error === 'invalid-format') {
+      return i18n('cc-domain-management.form.domain.error.format');
+    }
+    if (error === 'invalid-wildcard') {
+      return i18n('cc-domain-management.form.domain.error.wildcard');
+    }
+    if (error?.code != null && error?.code === 'hostname-contains-path') {
+      return i18n('cc-domain-management.form.domain.error.contains-path', { path: error.pathWithinHostname });
+    }
+    return null;
```

Comment:
well it wasn't possible because `error` could be a `string` or an object so typical `switch` was not really appropriate (or you'd have to use the `switch(true) ...` pattern which I personally don't like :roll_eyes: )

Since I've moved to errors always being objects with codes I can now use a switch as you suggest :+1: